yearn/prices/magic.py

```diff
@@ -156,22 +67,39 @@ def get_price_eth(token, block=None):
         price = balancer.get_price(token, block=block)
         logger.debug("balancer pool -> %s", price)
 
-    # peel a layer from [multiplier, underlying]
+    elif yearn_lens.is_yearn_vault(token):
```

Comment:
i guess if we change the contains behavior, we could move this below with other markets. what do you think?

Reply:
Went a step further and got rid of the `pool-like` markets and added two interface methods: `is_in_pool` and `get_pool_price`.
That way we don't need to forcefully change the semantics of yearn_lens's `__contains__` and also have the ability to check for token inclusion in a specific market + token inclusion in a specific market's pool like e.g. `uniswap_v2`

It's a bit ugly, maybe you have a better idea, I thought this would be a minimally invasive code change.

Reply:
reverted this for more readability as discussed